reactioncommerce/reaction
# User prefs stored in Accounts

### Closes #3404 

The issue is happening because of the following reason
1. Whenever some sub-menu is opened(or any action is taken), `Meteror.user().profile.preferences` is updated.
2. This causes all the `Tracker.autorun()` that depend on Meteor.user() to run again. This takes huge amount of time and is the main reason for the UI to be lagging.
An example would be the i18n startup script(https://github.com/reactioncommerce/reaction/blob/master/client/modules/i18n/startup.js#L41), which uses Meteor.user() and is triggered every time some thing is updated in `Meteor.user()`

To fix this, the user preference will be now be stored in`Accounts` collection.

### To test

1. Run `reaction`
2. Log in as admin, click on a product, notice the grid panel opens almost instantaneously.
3. Open social settings from Action menu, open/close some sub-menu. Notice there is no lag in opening of the menus now.
4. Take a note of the sub-menus, opened and closed.
5. Log out and Log in as admin
6. Reopen the social settings and only the same sub-menu should be open that were left opened before logout.
7. Change the shop from marketplace and then logout.
8. Browse to RC in incognito mode, login as admin.
9. The same shop should open that was opened on last login.

I have tried my best to check all the places where `preferences` are being used. If anyone can think of anything else, please let me know.
